### PR TITLE
Feature/questions

### PR DESCRIPTION
**Pull request by Alinaprotsyuk**
change Network manager for Subject entity and Question entity

### CODE REVIEW

File: DTAdmin/MainTabBarController.swift [hunk 1]
```diff
@@ -49,14 +49,14 @@ class MainTabBarController: UITabBarController, UITabBarControllerDelegate {
         tabBarViewControllers.append(specialityTab)
         
 //MARK: Create admins tab
-        let adminStoryboard = UIStoryboard.stoyboard(by: .admin)
+        let adminStoryboard = UIStoryboard(name: "Admin", bundle: nil)
```
Comment (iostaf) [inline DTAdmin/MainTabBarController.swift:52]: Shouldn't it be `UIStoryboard.storyboard(by: .admin)`? Please discuss it with Bohdan Hutsul

File: DTAdmin/MainTabBarController.swift [hunk 2]
```diff
@@ -49,14 +49,14 @@ class MainTabBarController: UITabBarController, UITabBarControllerDelegate {
...
         guard let adminsNavController = adminStoryboard.instantiateViewController(withIdentifier: "AdminListView") as? UINavigationController else { return }
         let adminsBarItem = UITabBarItem(title: "Admins", image: nil, selectedImage: nil)
         adminsNavController.tabBarItem = adminsBarItem
         tabBarViewControllers.append(adminsNavController)
         
 //MARK: Create timeTable tab
-        let timeTableStoryboard = UIStoryboard.stoyboard(by: .timeTable)
+        let timeTableStoryboard = UIStoryboard(name: "TimeTable", bundle: nil)
```
Comment (iostaf) [inline DTAdmin/MainTabBarController.swift:59]: Shouldn't it be UIStoryboard.storyboard(by: .timeTable)? Please discuss it with Bohdan Hutsul

File: DTAdmin/Models/DataManager.swift [hunk 3]
```diff
@@ -9,17 +9,17 @@
...
-/**
+    /**
      Returns an array containing the non-nil defined type of elements from API
      - Precondition: Get list all records is not supported by API for next enitities: __Student__, __Question__, __Answer__, __User__
      - Parameters:
-        - typeEntity : Type of entity from Entities enum
-        - listEntity : Returns an optional array of elements type Any. Use __cast to type__ for all elements according to structure of data
-        - error : Optional string in case of error while receiving data
-*/
+     - typeEntity : Type of entity from Entities enum
```
Comment (iostaf) [inline DTAdmin/Models/DataManager.swift:19]: Why have you removed leading spaces?

File: DTAdmin/Models/DataManager.swift [hunk 4]
```diff
@@ -105,7 +105,7 @@ class DataManager: HTTPManager {
                     }
                 }
             }
-        }.resume()
+            }.resume()
```
Comment (iostaf) [inline DTAdmin/Models/DataManager.swift]: Bad indentation

File: DTAdmin.xcodeproj/project.pbxproj [hunk 5]
```diff
@@ -1,702 +1,832 @@
...
-		C7535CF81FA2670100B6CBD1 /* TestStructure.swift in Sources */ = {isa = PBXBuildFile; fileRef = C7535CE71FA2670100B6CBD1 /* TestStructure.swift */; };
-		C7535CF91FA2670100B6CBD1 /* TimeTableStructure.swift in Sources */ = {isa = PBXBuildFile; fileRef = C7535CE81FA2670100B6CBD1 /* TimeTableStructure.swift */; };
-		C7535CFA1FA2670100B6CBD1 /* UserStructure.swift in Sources */ = {isa = PBXBuildFile; fileRef = C7535CE91FA2670100B6CBD1 /* UserStructure.swift */; };
-		C799D68C1FA267590092170E /* GroupTableViewCell.swift in Sources */ = {isa = PBXBuildFile; fileRef = C799D6891FA267580092170E /* GroupTableViewCell.swift */; };
-		C799D68D1FA267590092170E /* Student.storyboard in Resources */ = {isa = PBXBuildFile; fileRef = C799D68A1FA267580092170E /* Student.storyboard */; };
-		C799D68E1FA267590092170E /* StudentsTableViewCell.swift in Sources */ = {isa = PBXBuildFile; fileRef = C799D68B1FA267580092170E /* StudentsTableViewCell.swift */; };
+        3F67C9031FA9DFEF002DDA64 /* AddNewQuestionViewController.swift in Sources */ = {isa = PBXBuildFile; fileRef = 3F67C9021FA9DF85002DDA64 /* AddNewQuestionViewController.swift */; };
```
Comment (iostaf) [inline DTAdmin.xcodeproj/project.pbxproj]: What have you done to produce such indentation noise?

File: DTAdmin/Assets.xcassets/if_picture_1814111.imageset/Contents.json [hunk 6]
```diff
@@ -0,0 +1,21 @@
+{
+  "images" : [
+    {
+      "idiom" : "universal",
+      "filename" : "if_picture_1814111.png",
+      "scale" : "1x"
+    },
+    {
+      "idiom" : "universal",
+      "scale" : "2x"
```
Comment (iostaf) [inline DTAdmin/Assets.xcassets/if_picture_1814111.imageset/Contents.json]: Where are 2x and 3x assets?

File: DTAdmin/CommonNetworkManager.swift [hunk 7]
```diff
@@ -81,8 +81,6 @@ class CommonNetworkManager {
                         } catch {
                             print("Error json parsing")
                         }
-                    } else {
```
Comment (iostaf) [inline DTAdmin/CommonNetworkManager.swift:84]: Are you sure to remove this and below?

File: DTAdmin/Models/DataManager.swift [hunk 8]
```diff
@@ -166,6 +166,85 @@ class DataManager: HTTPManager {
...
+        let indexString = String(index)
+        let quantityString = String(quantity)
+        guard let request = self.getURLReqest(entityStructure: typeEntity, type: TypeReqest.GetRecordsRangeByTest, id: entityId, limit: quantityString, offset: indexString) else {
+            let error = "Cannot prepare header for URLRequest"
+            completionHandler(nil, error)
+            return
+        }
+        self.getResponse(request: request) { (list, error) in
+            if let error = error {
+                StoreHelper.logout()
```
Comment (iostaf) [inline DTAdmin/Models/DataManager.swift:179]: Why do we need this line?

File: DTAdmin/Models/DataManager.swift [hunk 9]
```diff
@@ -166,6 +166,85 @@ class DataManager: HTTPManager {
...
+                DispatchQueue.main.async {
+                    completionHandler(nil, error)
+                }
+            }
+            guard let  json = list as? [[String: Any]] else {
+                let error = NSLocalizedString("Response is empty", comment: "No data in server response")
```
Comment (iostaf) [inline DTAdmin/Models/DataManager.swift:185]: Bad error message. Consider a case when the server returns one element instead of an array.

File: DTAdmin/Student/Controllers/EditStudentViewController.swift [hunk 10]
```diff
@@ -191,13 +191,13 @@ class EditStudentViewController: UIViewController, UINavigationControllerDelegat
     }
 }
 extension UIImage {
-/**
+    /**
      Resize image to defined size and scale according to bitmap-based graphics context.
      - Parameters:
-        - size: The size (measured in points) of the new bitmap context.
-        - scale: The scale factor to apply to the bitmap.
+     - size: The size (measured in points) of the new bitmap context.
```
Comment (iostaf) [inline DTAdmin/Student/Controllers/EditStudentViewController.swift:196]: Why did you remove leading spaces?

File: DTAdmin/Models/DataManager.swift [hunk 11]
```diff
@@ -166,6 +166,85 @@ class DataManager: HTTPManager {
...
+            case .Student: entytiList = json.flatMap { StudentStructure(dictionary: $0) }
+            case .User: entytiList = json.flatMap { UserStructure(dictionary: $0) }
+            }
+            DispatchQueue.main.async {
+                completionHandler(entytiList, nil)
+            }
+        }
+    }
+    
+    func getEntityByAnother(byId: String, typeEntity: Entities, completionHandler: @escaping (_ entity: Any?, _ error: String?) -> ()) {
```
Comment (iostaf) [inline DTAdmin/Models/DataManager.swift]: What does this method do? Its name is very confusing for me.

File: DTAdmin/Models/DataManager.swift [hunk 12]
```diff
@@ -166,6 +166,85 @@ class DataManager: HTTPManager {
...
+                    completionHandler(nil, error)
+                }
+            }
+            guard let  json = list as? [[String: Any]] else {
+                print("Response is empty")
+                return
+            }
+            var entytiList = [Any]()
+            switch typeEntity {
+            case .Faculty: entytiList = json.flatMap { FacultyStructure(dictionary: $0) }
```
Comment (iostaf) [inline DTAdmin/Models/DataManager.swift]: I've seen it above. How about to extract repetitive code into a function?

File: SubjectDTAdminTests/SubjectDTAdminTests.swift [hunk 13]
```diff
@@ -0,0 +1,65 @@
...
+    }
+    
+    override func tearDown() {
+        sessionUnderTest = nil
+        super.tearDown()
+    }
+    
+    func testValidCallHTTPStatusCode200() {
+        // given
+        let url = URL(string: "https://vps9615.hyperhost.name/login/index")
```
Comment (iostaf) [inline SubjectDTAdminTests/SubjectDTAdminTests.swift]: Bad test. Unit tests have to test your code but not backend calls.

**Review by iostaf — CHANGES_REQUESTED** [review]

File: DTAdmin/CommonNetworkManager.swift (hunk 7 above)
Comment (Alinaprotsyuk) [reply DTAdmin/CommonNetworkManager.swift:84]: I didn't do anything here. it's not my file

File: DTAdmin/Models/DataManager.swift (hunk 9 above)
Comment (Alinaprotsyuk) [reply DTAdmin/Models/DataManager.swift:185]: He always return an array, in case with one element it will be array with one element

File: DTAdmin/Models/DataManager.swift (hunk 8 above)
Comment (Alinaprotsyuk) [reply DTAdmin/Models/DataManager.swift:179]: I think, if response has an error we want to logout from the program
